python/events/evaluation_tools/events/event_detection/decomposition.py

```diff
@@ -177,23 +176,23 @@ def mark_event_flows(
 
     # Eliminate negative values
     detrended[detrended < 0.0] = 0.0
-
-    # Do not filter events
-    if minimum_event_duration == '0H':
-        # Return mask of non-zero detrended flows
-        return pd.Series((detrended > 0.0), index=series.index)
     
     # Generate mask of non-zero detrended flows
     event_points = pd.Series((detrended > 0.0), index=series.index)
 
+    # Do not filter events
+    if minimum_event_duration == 0:
+        # Return mask of non-zero detrended flows
+        return event_points
+
     # Get list of potential events
     events = event_boundaries(event_points)
 
     # Compute durations
     durations = events['end'].sub(events['start'])
 
     # Filter events
-    events = events[durations >= to_offset(minimum_event_duration)].reset_index(drop=True)
+    events = events[durations >= pd.Timedelta(minimum_event_duration)].reset_index(drop=True)
```

Comment:
Still run into the issue with the `if minimum_event_duration == 0` conditional above. I think right before the conditional you should set `minimum_event_duration = pd.Timedelta(minimum_event_duration)`. Then the conditional should be `if minimum_event_duration == pd.Timedelta("0H")`

Reply:
Implemented.